Update validationDefault

yosus opened the pull request:
Updated  validationDefault from protected to public

yosus committed:
Update validationDefault

Updated  validationDefault from protected to public

yosus committed:
Added with getData() sample

added getData() sample, so users will know about this.

othercorey commented on en/core-libraries/form.rst:
```diff
@@ -130,6 +130,9 @@ Values set with this method will overwrite existing data in the form object::
                     'email' => 'john.doe@example.com'
                 ]);
+                // Get the values like this
+                $allValues = $contact->getData();
```
I don't know if this example makes sense to put here. It's not related to the rest of the code.

markstory replied:
It could be in a separate example?

othercorey replied:
I would make them separate if it's important to include.

yosus replied:
ok, sure. let me create a separate getData sample.

yosus committed:
Update form.rst

Updated minLength to 5, because the example 'John Doe' is only 8 in length.

yosus reviewed (commented):
Updated minLength in validator
Added getting-data seperate sample
I would like to note that in cake 3.9.6, calling `execute` does not automatically save the data. After cake 4.1.0, calling `execute` will automatically save the data.

Current doc flow is :

1. Create Form
2. **Processing Data**
3. Setting Data
4. Getting Data
5. Getting Errors


Would it be better to put Processing-Data nearer Getting-Errors. Like this :

1. Create Form
2. Setting Data
3. Getting Data
4. **Processing Data**
5. Getting Errors

markstory commented:
> Would it be better to put Processing-Data nearer Getting-Errors.

That sounds like a better structure to me :+1:

yosus committed:
Relocate 'Processing Request Data'

Relocate 'Processing Request Data' section so it is before 'Getting Form Errors' section.

yosus reviewed (commented):
Relocated `Processing Request Data` section

markstory commented:
Thank you 👍
